### PR TITLE
Annotate for nullable reference types

### LINKED ISSUE #88: Annotate system/plugin API
- https://www.nuget.org/packages/JetBrains.Annotations/ will be useful to get <code>@Nonnull</code> and <code>@CheckForNull</code> annotations
- other annotations like <code>@Extension</code> could be useful for extensions engine

### PR DESCRIPTION
**Pull request by NextTurn**
Closes #88 
Based on #354 

It's unclear to me how to annotate those properties. See suppressed warnings in https://github.com/kohsuke/winsw/commit/31d1d134264e6e2bedfc1659dd8773cfdc82d726, https://github.com/kohsuke/winsw/commit/6d57da2c92fce41cc917c8881dd8391a1b3d8c00.

### CODE REVIEW

**Comment by NextTurn** [comment]
Please don't add #88 to the title... This is not to fix that but it closes that by the way as a superseded/stale one.

File: src/Core/ServiceWrapper/Main.cs [hunk 1]
```diff
@@ -427,16 +428,11 @@ private void StartProcess(Process processToStart, string arguments, string execu
                 {
                     LogEvent("WaitForExit " + ioe.Message);
                 }
-
-                try
+                finally
                 {
                     proc.Dispose();
                 }
-                catch (InvalidOperationException ioe)
```
[the remark this answers is not in the document]
Comment (NextTurn) [reply src/Core/ServiceWrapper/Main.cs]: I did make some unrelated minor changes. For this specific case, `Process.Dispose` will not throw.

**Comment by oleg-nenashev** [comment]
Will need a merge conflict resolution before merging. Chained PRs are not fun to maintain

**Comment by NextTurn** [comment]
> Will need a merge conflict resolution before merging.

I'll rebase this.

> Chained PRs are not fun to maintain

Hopefully this is the last one.

**Pull request merged**